iroh-net/examples/magic.rs

```diff
@@ -52,9 +52,7 @@ async fn main() -> anyhow::Result<()> {
 
     let derp_map = match args.derp_url {
         None => default_derp_map(),
-        Some(url) => {
-            DerpMap::default_from_node(url, DEFAULT_DERP_STUN_PORT, UseIpv4::None, UseIpv6::None)
-        }
+        Some(url) => DerpMap::default_from_url(url),
```

Comment:
I wonder if we should use `From<Url>` instead

Reply:
I find `From` instances sometimes a bit hard to discover for such things.

Reply:
I agree with `From<Url>` though, it's very much the interface you expect from rust.  You can help discoverability, but also once you know about `From` you know to go look for it and really do expect it for any conversion that does not require extra input.

Reply:
Maybe let's just do both? `From<Url>` but also keep `DerpMap::default_from_url` as public method?

Reply:
Why keep both? The `From` impl is public as well and shows up nicely in the docs

Reply:
I was wondering about docs to note that a DNS lookup will be performed and default STUN port will be assumed. There's no way to add docs to trait impls, right?
Maybe also here it doesn't matter much. Your call :)